packages-exp/auth-exp/src/api/authentication/token.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { querystring } from '@firebase/util';
+
+import { performFetchWithErrorHandling } from '../';
+import { Auth } from '../../model/auth';
+
+export const _ENDPOINT = 'https://securetoken.googleapis.com/v1/token';
+const GRANT_TYPE = 'refresh_token';
+
+enum ServerField {
+  ACCESS_TOKEN = 'access_token',
+  EXPIRES_IN = 'expires_in',
+  REFRESH_TOKEN = 'refresh_token'
+}
+
+export interface RequestStsTokenResponse {
+  accessToken?: string;
+  expiresIn?: string;
+  refreshToken?: string;
+}
+
+export async function requestStsToken(
+  auth: Auth,
+  refreshToken: string
+): Promise<RequestStsTokenResponse> {
+  const response = await performFetchWithErrorHandling<{
+    [key: string]: string;
```

Comment:
should this be a named type instead of just a string map?

Reply:
Can't-- ts lint stomps its feet and throws a tantrum if you have snake_case members/variables

Reply:
you can ignore es-lint for that interface `eslint-disable-line`

Reply:
Ya it was gonna be like ten different places for that though, but no matter I figured out how to disable it for the whole file